libraries/chain/controller.cpp

```diff
@@ -1388,7 +1395,7 @@ fork_database& controller::fork_db()const { return my->fork_db; }
 
 void controller::start_block( block_timestamp_type when, uint16_t confirm_block_count) {
    validate_db_available_size();
-   my->start_block(when, confirm_block_count, block_status::incomplete );
+   my->start_block(when, confirm_block_count, block_status::incomplete, block_id_type() );
```

Comment:
With the change to `optional<block_id_type>`, the last argument of this function call would be `optional<block_id_type>()` instead of `block_id_type()`.